### DISCUSSION TITLE
how can I find out the maximum column values in a table?

### DISCUSSION THREAD

**Discussion by IknowL1f3** [supabase-community/supabase-csharp#155] [Q&A]
I found how it's done in JS, but there's nothing in C# (or maybe I didn't see it). How can I find the maximum value of a field in a column?

js:
const { data, error } = await supabase
  .from('movies')
  .select('release_year.max(), release_year.min()')

**Comment by acupofjose** [comment]
You can do the same thing in the C# client using `Select`! 

The difference is you would need to alias the Selects (`release_year.max AS Max`), create a `Model` (that inherits from `BaseModel`) to hydrate the data into, and then add `Column` attributes for the aliased Select values (`[Column(“Max”)] public int Max { get; set}`)

**Reply by IknowL1f3** [reply]
I didn't quite understand you, I have a request to get the number of records: "Console.WriteLine(await supabase.From<Balance>().Select(x => new object[] { x.Id }).Count(Supabase.Postgrest.Constants.CountType.Exact));", I don't want to find out their number, but the maximum value, how can i do this?

**Reply by acupofjose** [reply]
One approach would be to model your query so that you can work with the hydrated values like this:

`MoviesYearMinMax.cs`
```c#
[Table("movies")]
public class MoviesYearMinMax : BaseModel 
{  
   [Column("max")] // <--- Mirrors the alias of the "column" from the SELECT statement
   public int Max { get; set; }

   [Column("min")]
   public int Min { get; set; }
}
```

Then do your query as follows:
```c#
const result = await supabase.From<MoviesYearMinMax>() // Note: Model that we're hydrating into
          .Select('release_year.max() as max, release_year.min() as min') // Note: the alias here matches the model
          .Single();
```
